src/features/web-compat.js

```diff
@@ -503,6 +503,11 @@ export default class WebCompat extends ContentFeature {
     }
 
     viewportWidthFix () {
+        if (document.readyState === 'loading') {
+            // if the document is not ready, we may miss the original viewport tag
+            document.addEventListener('DOMContentLoaded', () => this.viewportWidthFix())
```

Comment:
I'd rename the rest of the method `viewportWidthFixInner` or similar and have the remainder of this function and this listener call that.

So this function would become:
```javascript
    viewportWidthFix() {
       if (document.readyState === 'loading') {
            // if the document is not ready, we may miss the original viewport tag
            document.addEventListener('DOMContentLoaded', () => this.viewportWidthFixInner())
            return
       }
       this.viewportWidthFixInner()
    }
```

This reduces the risk of edge case re-entrancy problems in future.

Alternatively you could solve it at the current `viewportWidthFix` call site but I think that isn't quite as clean.